patches/web_contents_impl.patch

```diff
@@ -1,8 +1,24 @@
 diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
-index b5b0b78..aac6876 100644
+index b5b0b78..1c42b6c 100644
 --- a/content/browser/web_contents/web_contents_impl.cc
 +++ b/content/browser/web_contents/web_contents_impl.cc
-@@ -4353,6 +4353,7 @@ void WebContentsImpl::RendererUnresponsive(
+@@ -1820,10 +1820,14 @@ void WebContentsImpl::CreateNewWindow(
+       static_cast<SessionStorageNamespaceImpl*>(session_storage_namespace);
+   CHECK(session_storage_namespace_impl->IsFromContext(dom_storage_context));
+ 
++  WindowContainerType window_container_type = WINDOW_CONTAINER_TYPE_NORMAL;
++  if (params.disposition == NEW_BACKGROUND_TAB)
```

Comment:
Yup true, in electron atleast currently we dont rely on the content layer to create the webContents. `AddNewContents` wont get triggered unless you allow the `content_layer` to create and `CanCreateWindow` is useless to associate with the webcontents we are gonna create, as the frame_id are gonna be obselete once we return `false` in `ShouldCreateWebContents`. 

If `ShouldCreateWebContents` provided the navigation policy, this patch woulnt have been required, but if were to add another param to the method I would be changing a few other files which is really unnecessary and will become hard to maintain.

Associating `WindowContainerType` and `disposition` are wrong, but this is the minimal way i could send across the navigation policy to dispatch in `new-window` event.